libs/execution_context/ebpf_native.c

```diff
@@ -921,16 +921,22 @@ _ebpf_native_set_initial_map_values(_Inout_ ebpf_native_module_t* module)
             }
 
             ebpf_handle_t handle_to_insert = ebpf_handle_invalid;
-
-            if (native_map_to_update->entry->definition.type == BPF_MAP_TYPE_ARRAY_OF_MAPS) {
+            ebpf_map_definition_in_file_t map_def = native_map_to_update->entry->definition;
+            uint32_t key_size = map_def.key_size;
+            bool is_hash_of_maps_with_int_key = (map_def.type == BPF_MAP_TYPE_HASH_OF_MAPS) &&
+                                                (key_size == sizeof(uint8_t) ||
```

Comment:
what if key size is uint8_t and max_entries exceeds 256 etc.? Does this code handle that? @Alan-Jowett  can you confirm if Linux mandates key_size to be strictly 4-bytes in this case?

Reply:
Can you please clarify the concern on the max_entries and key_size set?

Reply:
@shankarseal : Could you please help clarify the discussed suggestion for the value_size?

For static initialization of hash-of-maps,
1. Add key_size == sizeof(int) or sizeof(long)  (so, it is only 4 bytes allowed).
2. value_size == ???

Reply:
With the suggested changes of key_size == sizeof(int), adding value into the inner map array is failing. Hence the suggestion does not work.

Reply:
If the key size is x, there can be at most 2^8x entries in the map which must be checked.
Also, it is not enough to only check the length of the key; but the type should be an integer. Not sure if/how that can be verified here.

Reply:
If bpf2c has that info (from parsing btf data), I think we need to change bpf2c to add that information in native driver, and then this code can check it.